Sources/Source/Components/ScrollingPaginationIndicator.swift

```diff
@@ -0,0 +1,125 @@
+        self.indicatorWidth = indicatorWidth
+        self.spacing = spacing
+        self.primaryColor = primaryColor
+        self.secondaryColor = secondaryColor
+        self._selectedIndex = selectedIndex
+    }
+
+    /// Calculates the scale factor for an item based on its distance from the selected index.
+    ///
+    /// The scaling effect is such that items closer to the selected index are
+    /// scaled larger, while those farther away are scaled smaller. The scale factor is clamped between
+    /// `minimumScale` and `maximumScale`.
+    ///
+    /// - Parameter index: The index of the item for which the scale factor is to be calculated.
+    /// - Returns: A `CGFloat` value representing the scale factor for the item at the given index.
+    private func scale(for index: Int) -> CGFloat {
+        guard pageCount >= numberOfVisibleDots else { return 1.0 }
+        let indexDifference = abs(index - selectedIndex)
+        let scaleSpread = max((CGFloat(numberOfVisibleDots - 1) / 2 + 1), 1)
+        let scaleFactor = CGFloat(indexDifference) / scaleSpread
+        return max(1.0 + scaleFactor * (minimumScale - 1.0), minimumScale)
+    }
+
+    public var body: some View {
+        ScrollViewReader { scrollViewProxy in
+            ScrollView(.horizontal) {
+                HStack(spacing: spacing) {
+                    ForEach(0..<pageCount, id: \.self) { index in
+                        Circle()
+                            .id(index)
+                            .foregroundStyle(selectedIndex == index ? primaryColor : secondaryColor)
+                            .frame(width: indicatorWidth)
+                            .scaleEffect(scale(for: index))
+                            .animation(.smooth, value: selectedIndex)
+                    }
+                }
+            }
+            .disabled(true) // The scroll view is only used for the scroll effect and shouldn't be interactable
+            .frame(width: scrollViewWidth)
+            .onChange(of: selectedIndex) { newValue in
+                withAnimation {
+                    scrollViewProxy.scrollTo(newValue, anchor: .center)
+                }
+            }
+        }
+    }
+}
+
+struct ScrollingPageIndicator_Previews_Container: PreviewProvider {
+    struct Container: View {
+        @State var selectedIndex = 0
+        let elementArray = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+        var body: some View {
+            VStack {
+                TabView(selection: $selectedIndex) {
+                    ForEach(elementArray, id: \.self) { index in
+                        Text("\(index)")
+                            .font(.largeTitle)
+                    }
+                }
+                .tabViewStyle(.page(indexDisplayMode: .never))
+                ScrollingPaginationIndicator(pageCount: elementArray.count, indicatorWidth: 16, selectedIndex: $selectedIndex, primaryColor: .blue, secondaryColor: .black)
```

Comment:
I can see the figma designs don't seem to have any colour detail (it's all just pink!) but I wonder if it is worth clarifying with design if we want to use a couple of dynamic colours from the design system? Ie a Guardian palette "blue" instead of just Apple's `.blue`.

Reply:
Good point - I've just used these colours because this is just a preview of the component. 
But given we've got access to the Guardian palette colours in the source repo I might just update anyway to make the previews nicer :)